src/api/api.tsx

```diff
@@ -0,0 +1,121 @@
+const BASE_URL = "https://panda-market-api.vercel.app/products";
+
+//Item 타입 정의
+export type Item = {
+  id: number;
+  name: string;
+  price: number;
+  description: string;
+  tags: string[];
+  images?: string[];
+  ownerId: number;
+  ownerNickname: string;
+  favoriteCount: number;
+  createdAt: string;
+};
+
+// getItems 타입 설정
+type GetItemsParams = {
+  page?: number;
+  pageSize?: number;
+  orderBy?: string;
+  keyword?: string;
+};
+
+// getItems 함수 반환 타입 설정
+type GetItemsResponse = {
+  totalCount: number;
+  list: Item[];
+};
+
+export async function getItems({
+  orderBy = "",
+  pageSize = 1,
+  keyword = "",
+  page = 1,
+}: GetItemsParams): Promise<GetItemsResponse> {
+  const query = `page=${page}&orderBy=${orderBy}&pageSize=${pageSize}&keyword=${keyword}`;
+  const response = await fetch(`${BASE_URL}?${query}`);
+  if (!response.ok) {
+    throw new Error("아이템을 불러오는데 실패했습니다.");
+  }
+  const body = await response.json();
+  return {
+    list: body.list,
+    totalCount: body.totalCount,
+  };
```

Comment:
코드가 조금 반복적으로 느껴져요! 
다음과 같이 해보면 어떨까요? 

```suggestion
  const data = await response.json();
  return data
```

또는 
```suggestion
  const { list, totalCount } = await response.json();
  return { list, totalCount };
```